src/kirby/components/modal/services/modal-service.tns.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { ModalDialogService } from 'nativescript-angular/directives/dialogs';
+import { ViewContainerRef } from '@angular/core';
+
+import { ModalComponent } from '~/kirby/components/modal/modal.component';
+import { ModalConfig } from '~/kirby/components/modal/config/modal-config';
+import { ModalCloserService } from './modal-closer-service';
+import { ModalServiceInterface } from './modal-service-interface';
+
+@Injectable()
+export class ModalService implements ModalServiceInterface {
```

Comment:
Consider of the modal service could have only one implementation (by splitting the service we have duplicated logic, e.g. regarding `uid`) and the differences abstracted away like: https://docs.nativescript.org/code-sharing/code-splitting#helper-file

Reply:
I agree that we can reuse uid, but imo in this particular case this change adds more overhead than it helps us reuse code. Despite that, I still implemented it, as we can now also return the uid in in its plain type (which i also think is unnecessary, but check my other comments)